### PR TITLE
Right now, those PMs are silently discarded.

### PR DESCRIPTION
**Pull request by Themroc**

### CODE REVIEW

**Comment by davidgfnet** [comment]
So the fact that I'm not seeing any encoded messages in my sniffing is because when I activated my phone number I did not specify an Android phone. Is that right?

File: README.md [hunk 1]
```diff
@@ -66,6 +66,26 @@ In order to get a proper backtrace, you can either use your package manager to
 install a debug package or rebuild the library with debug symbols using `make
 debug`.
 
+### Why are some private messages not displayed?
+
+Since the end of 2014, wahtsapp's Android clients feature end-to-end encryption
```
Comment (dequis) [inline README.md]: u waht m8

**Comment by Themroc** [comment]
Probably. It appears like they switch on this encryption only when registering with resource saying "Android after ca. dec 2014" (and from what i've seen on the net, their iphone client doesn't support it at all). At least, thats what my experiments suggest. First, i registered my landline with a pretty recent yowsup and was wondering why i got groupchat replys, but no PMs. Then, after some investigations, i registered a virtual number using a patched yowsup as described. I also set the default_resource[] in wa_purple.c to 2.11.414 and renamed ~/.purple, to have a fresh start. Bingo, now i could see PMs coming in. Even after i changed the resource string in whatsapp-purple back to 2.12.5, it still worked :-/. But maybe the reason was not the version string, but the _AXOLOTL = False. I can't try that, since i've used up all available phone numbers... At any rate, with a fresh phone number and a recent unpatched yowsup you should get encrypted PMs.

**Comment by Themroc** [comment]
Gr8, http://phone.ipkall.com/ is working again. I've just got a second virtual phone no., gonna try with _VERSION = "2.11.471" and _AXOLOTL = True...

**Comment by davidgfnet** [comment]
Hey. Im expericencing this now. Since last week.
I see composing messages but I dont see any other message between composing and paused.
Ill take a look using wireshark cause pidgin in debug mode is not dumping any frame.

**Comment by davidgfnet** [comment]
Nevermind, just changed resource to 2.11.414 and reconnected: received all the old messages peding. After that deleted resource (to go to default) and now it works. Im pretty sure I know what it is, just let me check.

**Comment by davidgfnet** [comment]
Just did a push, can you check?

**Comment by Themroc** [comment]
No luck. With both accounts, i see "Received message stanza..." in debug-output, but the one registered without _AXOLOTL = False is still encrypted, so nothing comes out in the chat window.
On a third account, registered with _VERSION = "2.11.471" and  _AXOLOTL = False, i get unencrypted PMs, so it looks like only the  _AXOLOTL = False matters.

**Comment by davidgfnet** [comment]
Could you compile as debug (make debug) and post some of the frames you get? That would be really helpful!

**Comment by dequis** [comment]
Merging this can't hurt, right? At least the first commit. The second one feels like 'giving up', heh.

**Comment by davidgfnet** [comment]
Yep sorry, I'm a bit off reviewing pull requests. I'll merge them at some point :)

**Pull request closed without merging**